databricks_cli/secrets/cli.py

```diff
@@ -0,0 +1,312 @@
+        ret.append((truncate_string(s['name']), s['backend_type']))
+    return ret
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Lists all secret scopes.')
+@click.option('--output', help=OutputClickType.help, type=OutputClickType())
+@profile_option
+@eat_exceptions
+@provide_api_client
+def list_scopes(api_client, output):
+    """
+    Lists all secret scopes.
+    """
+    scopes_json = SecretApi(api_client).list_scopes()
+    if OutputClickType.is_json(output):
+        click.echo(pretty_format(scopes_json))
+    else:
+        click.echo(tabulate(_scopes_to_table(scopes_json), headers=SCOPE_HEADER))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Deletes a secret scope.')
+@click.option('--scope', required=True, type=SecretScopeClickType(), help=SecretScopeClickType.help)
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_scope(api_client, scope):
+    """
+    Deletes a secret scope.
+    """
+    SecretApi(api_client).delete_scope(scope)
+
+
+def _verify_and_translate_options(string_value, binary_file):
+    """
+    Translates options into actual parameters for API call.
+    Return tuple with two values representing (string_value, bytes_value).
+    """
+    if string_value and binary_file:
+        error_and_quit("At most one of {} should be provided."
+                       .format(['string-value', 'binary-file']))
+
+    elif string_value is None and binary_file is None:
+        prompt = '# Remove this line and input your secret value.' + \
+            ' Any trailing new line will be stripped and text will be stored in' + \
+            ' UTF-8 (MB4) form. Exit without saving to abort writing secret.'
+
+        # underlying edit function made sure using a temporary file for editing
+        content = click.edit(prompt)
+        # return None means editor is closed without changes
+        if content is None:
+            error_and_quit('No changes made, write secret aborted.'
+                           ' Please follow the instruction to input secret value.')
+
+        return content.rstrip('\n'), None
+
+    elif string_value is not None:
+        return string_value, None
+
+    elif binary_file is not None:
+        with open(binary_file, 'rb') as f:
+            binary_content = f.read().rstrip('\n')
+
+        if not is_base64_str(binary_content):
+            error_and_quit("Content in file {} is probably not base64 encoded.".format(binary_file))
+
+        return None, binary_content
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Writes a secret to a scope.')
+@click.option('--scope', required=True, type=SecretScopeClickType(), help=SecretScopeClickType.help)
+@click.option('--key', required=True, type=SecretKeyClickType(), help=SecretKeyClickType.help)
+@click.option('--string-value', default=None,
+              help='Read value from string and stored in UTF-8 (MB4) form')
```

Comment:
Just to confirm, do we do any encoding/transformation?

Reply:
I think this is an extra warning as a comparison to the `--binary-file` option. If user stores bytes value, `dbutils.preview.secret.get` may not return a string.

Reply:
@mengxr It's important for users to know that if they store their secrets as strings in the backend, we assume an encoding for when it's subsequently read via `dbutils.preview.secret.get`. Depending on the user-defined encoding, we may not be able to provide them a reliable encoding, so if they want to handle encoding themselves, they must use --binary-file.